glommio/src/sys/mod.rs

```diff
@@ -487,6 +488,8 @@ impl Wakers {
     }
 }
 
+pub(crate) type StatsCollectionFn = fn(&io::Result<usize>, &mut RingIoStats) -> ();
```

Comment:
this is a bit un-rusty, but should be okay for now as I am interested in merging this soon.

The reason is, `fn` cannot capture state, so it is rarely used. More often you will see `Fn` and `FnMut`, but since those are traits, they require a Box to be stored.

Reply:
Yes this is done on purpose. Using a `Fn` (or variant) would add an direction (via `dyn`) and an extra allocation. Because I don't need to capture state, this is actually very simple and efficient.